File: src/test/java/kimspring/splearn/application/member/provided/MemberFinderTest.java
package kimspring.splearn.application.member.provided;

import jakarta.persistence.EntityManager;
import jakarta.transaction.Transactional;
import kimspring.splearn.SplearnTestConfiguration;
import kimspring.splearn.domain.member.Member;
import kimspring.splearn.domain.member.MemberFixture;
import org.junit.jupiter.api.Test;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.context.annotation.Import;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

@Transactional
@SpringBootTest
@Import(SplearnTestConfiguration.class)
record MemberFinderTest(
        MemberFinder memberFinder,
        MemberRegister memberRegister,
        EntityManager entityManager
) {

    @Test
    void find() {
        Member member = memberRegister.register(MemberFixture.createMemberRegisterRequest());
        entityManager.flush();
        entityManager.clear();

        Member found = memberFinder.find(member.getId());

        assertThat(member.getId()).isEqualTo(found.getId());
    }

    @Test
    void findFail() {
        assertThatThrownBy(() -> memberFinder.find(9999L))
            .isInstanceOf(IllegalArgumentException.class);
    }
}

File: src/test/kotlin/kimspring/splearn/application/member/usecase/MemberFinderTest.kt
package kimspring.splearn.application.member.usecase

import io.kotest.assertions.throwables.shouldThrow
import io.kotest.core.spec.style.FunSpec
import io.kotest.extensions.spring.SpringExtension
import io.kotest.matchers.shouldBe
import kimspring.splearn.domain.member.MemberFixture
import kimspring.splearn.domain.member.MemberNotFoundException
import kimspring.splearn.support.stereotype.ApplicationServiceTest
import org.springframework.beans.factory.annotation.Autowired

@ApplicationServiceTest
class MemberFinderTest : FunSpec() {
    @Autowired
    private lateinit var memberFinder: MemberFinder

    @Autowired
    private lateinit var memberRegister: MemberRegister

    init {
        extension(SpringExtension())

        test("get") {
            val member = memberRegister.register(MemberFixture.createRegisterMemberCommand())

            val found = memberFinder.get(member.id!!)

            member.id shouldBe found.id
        }

        test("getFail") {
            shouldThrow<MemberNotFoundException> { memberFinder.get(9999L) }
        }
    }
}
